### PR TITLE
feat: allow users to replace auth methods

### PR DESCRIPTION
**Pull request by meetcshah19**
# Description 📣
Allow users to replace / overwrite auth methods for identities.

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Comment by dangtony98** [comment]
Note: Instead of passing `initialAuthMethod` prop into each Form component, you can also get the it through `useGetIdentityById` in the form to get the auth method since it is specified on the identity.

I do think that this PR may conflict with the initiative to add multiple auth methods per identity @DanielHougaard

**Comment by DanielHougaard** [comment]
We're leaving this PR as-is for now because we have another initiative that will take precedence over this.

File: frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx
Comment (scott-ray-wilson) [inline frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx:1]: When you open this modal, change the selected auth method and close it again, and re-open it, it doesn't reset back to the currently used auth method, I think this should reset on close

**Review by scott-ray-wilson — CHANGES_REQUESTED** [review]

File: frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx
Comment (scott-ray-wilson) [inline frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx:1]: Also seems like the general logic of this whole flow isn't working?

image

**Review by scott-ray-wilson — CHANGES_REQUESTED** [review]

File: frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx
Comment (scott-ray-wilson) [reply frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx:1]: @meetcshah19 this error was due to having the confirm overwrite modal inside the form, the button click on the form triggered the submit on the form as well

**Review by scott-ray-wilson — APPROVED** [review]

**Review by maidul98 — APPROVED** [review]
tested it out, works as expected scott and meet

**Pull request merged**